### ISSUE TITLE
SIGSEGV: segmentation violation when trying to use a custom config

### ISSUE THREAD

**Issue by lucazz** [aquasecurity/kube-bench#339]
Hello everyone, 

I'm trying to override a couple of those checks on our gke clusters and am getting these errors when doing so: 
```
~ $ kubectl logs kube-bench-4kc5m
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x7d7768]

goroutine 1 [running]:
github.com/spf13/viper.(*Viper).find(0x0, 0x969617, 0xa, 0xa, 0x30)
        /go/pkg/mod/github.com/spf13/viper@v1.0.0/viper.go:865 +0x38
github.com/spf13/viper.(*Viper).Get(0x0, 0x969617, 0xa, 0xc0000dfbb8, 0x40ccad)
        /go/pkg/mod/github.com/spf13/viper@v1.0.0/viper.go:594 +0x7c
github.com/spf13/viper.(*Viper).GetStringSlice(0x0, 0x969617, 0xa, 0xc0000a01e0, 0x96797c, 0x6)
        /go/pkg/mod/github.com/spf13/viper@v1.0.0/viper.go:691 +0x3f
github.com/aquasecurity/kube-bench/cmd.getBinaries(0x0, 0x96797c, 0x6, 0x0)
        /go/src/github.com/aquasecurity/kube-bench/cmd/util.go:85 +0x66
github.com/aquasecurity/kube-bench/cmd.isMaster(0xc0000a14a0)
        /go/src/github.com/aquasecurity/kube-bench/cmd/common.go:237 +0x8d
github.com/aquasecurity/kube-bench/cmd.glob..func4(0xd85320, 0xdab9d8, 0x0, 0x0)
        /go/src/github.com/aquasecurity/kube-bench/cmd/root.go:60 +0x26
github.com/spf13/cobra.(*Command).execute(0xd85320, 0xc00001e270, 0x0, 0x0, 0xd85320, 0xc00001e270)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.1/command.go:702 +0x289
github.com/spf13/cobra.(*Command).ExecuteC(0xd85320, 0xdab9d8, 0x0, 0x0)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.1/command.go:783 +0x2ca
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.1/command.go:736
github.com/aquasecurity/kube-bench/cmd.Execute()
        /go/src/github.com/aquasecurity/kube-bench/cmd/root.go:75 +0x95
main.main()
        /go/src/github.com/aquasecurity/kube-bench/main.go:22 +0x20
~ $
```
Here's the kubernetes manifest I'm trying to run: 
https://gist.github.com/lucazz/6e5ecf171625b3ced776c5f31f43a1d1

What am I missing here?

Cheers,

**Comment by yoavrotems** [comment]
How are you trying to override those commands?

**Comment by azarzar15** [comment]
Hi,

im getting same error when compiling the binary on my ubuntu and trying to run it on the k8s master (debian).
i cant compile it on the debian itself...
is there a way maybe to compile it statically?

**Comment by lizrice** [comment]
My guess is that the seg violation error is to do with reading the test config from a config map rather than a file (see the gist from @lucazz). We shouldn't crash, that's definitely a bug!

@azarzar15 you should be able to compile the executable simply with `make build`

**Comment by davidmontoyago** [comment]
We are experiencing the same issue when loading custom config from a configmap with`kube-bench v0.0.33`.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x7d7778]

goroutine 1 [running]:
github.com/spf13/viper.(*Viper).find(0x0, 0x96a61d, 0xa, 0xa, 0x30)
        /go/pkg/mod/github.com/spf13/viper@v1.0.0/viper.go:865 +0x38
```

**Comment by lizrice** [comment]
The crash is (of course) a bug; reading the config from a configmap rather than failing gracefully would be an enhancement, but it looks like there's some support for the idea :-)

**Comment by simar7** [comment]
I'm kicking the tires with this as I write. Hopefully I'll have an update on this soon.

**Comment by simar7** [comment]
Hey @lucazz I had some time to look into this and the problem seems to be two-fold. Let me describe it in more detail:

1. It seems that you're trying to override the `node.yaml` by passing it in through the `ConfigMap` -– this is acceptable but there were a few issues with your yaml. Here's an updated gist with what we suspect you're trying to accomplish: [link](https://gist.github.com/simar7/3f64d69c35f0138b845422d9d8a5db76). Do note that the `mountPath` to `kube-bench-node` was fixed and also the key to the `ConfigMap`.

2. As a result of digging into this code, I found there's a bug which makes `kube-bench` crash. I'll be cutting a PR for that shortly.

Let us know if this helps or if we can further assist you. 
-s
